src/stack/Stack.h

```diff
@@ -0,0 +1,264 @@
+ */
+
+#pragma once
+
+#include <stack/Stack.h>
+#include <stack/Traits.h>
+
+#include <messaging/ExchangeMgr.h>
+#include <protocols/secure_channel/MessageCounterManager.h>
+#include <transport/SecureSessionMgr.h>
+#include <transport/TransportMgr.h>
+#include <transport/raw/UDP.h>
+
+#if CONFIG_DEVICE_LAYER
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/PlatformManager.h>
+#endif
+
+namespace chip {
+
+/* ========================== SystemLayer Configuration ========================== */
+struct SystemLayerConfiguration
+{
+};
+template <typename T>
+struct IsSystemLayerConfiguration : std::is_base_of<SystemLayerConfiguration, T>
+{
+};
+
+#if CONFIG_DEVICE_LAYER
+class DefaultSystemLayer : SystemLayerConfiguration
+{
+public:
+    CHIP_ERROR Init() { return CHIP_NO_ERROR; }
+    CHIP_ERROR Shutdown() { return CHIP_NO_ERROR; }
+
+    System::Layer & Get() { return DeviceLayer::SystemLayer; }
+};
+#else
+class DefaultSystemLayer : SystemLayerConfiguration
+{
+public:
+    CHIP_ERROR Init() { return mSystemLayer.Init(nullptr); }
+
+    CHIP_ERROR Shutdown() { return mSystemLayer.Shutdown(); }
+
+    System::Layer & Get() { return mSystemLayer; }
+
+private:
+    System::Layer mSystemLayer;
+};
+#endif // CONFIG_DEVICE_LAYER
+
+/* ========================== InetLayer Configuration ========================== */
+struct InetLayerConfiguration
+{
+};
+template <typename T>
+struct IsInetLayerConfiguration : std::is_base_of<InetLayerConfiguration, T>
+{
+};
+
+#if CONFIG_DEVICE_LAYER
+class DefaultInetLayer : InetLayerConfiguration
+{
+public:
+    CHIP_ERROR Init(System::Layer & aSystemLayer) { return CHIP_NO_ERROR; }
+    CHIP_ERROR Shutdown() { return CHIP_NO_ERROR; }
+
+    Inet::InetLayer & Get() { return DeviceLayer::InetLayer; }
+};
+#else
+class DefaultInetLayer : InetLayerConfiguration
+{
+public:
+    CHIP_ERROR Init(System::Layer & aSystemLayer) { return mInetLayer.Init(aSystemLayer, nullptr); }
+
+    CHIP_ERROR Shutdown() { return mInetLayer.Shutdown(); }
+
+    Inet::InetLayer & Get() { return mInetLayer; }
+
+private:
+    Inet::InetLayer mInetLayer;
+};
+#endif // CONFIG_DEVICE_LAYER
+
+/* ========================== Storage Configuration ========================== */
+struct StorageConfiguration
+{
+};
+template <typename T>
+struct IsStorageConfiguration : std::is_base_of<StorageConfiguration, T>
+{
+};
+
+class DefaultStorage : StorageConfiguration
+{
+public:
+    CHIP_ERROR Init() { return CHIP_NO_ERROR; }
+    CHIP_ERROR Shutdown() { return CHIP_NO_ERROR; }
+
+    PersistentStorageDelegate * Get() { return nullptr; }
+};
+
+/* ========================== BleLayer Configuration ========================== */
+struct BleLayerConfiguration
+{
+};
+template <typename T>
+struct IsBleLayerConfiguration : std::is_base_of<BleLayerConfiguration, T>
+{
+};
+
+class DefaultBleLayer : BleLayerConfiguration
+{
+public:
+    template <typename T>
+    CHIP_ERROR Init(T * stack)
+    {
+        return CHIP_NO_ERROR;
+    }
```

Comment:
sholuld we have a Shutdown here as well so that we do not need to go through BLE nullptr checks during shutdown?